tests/test_parser.py

```diff
@@ -86,6 +86,35 @@ def test_get_baseimg_from_df(self, dfparser):
         base_img = dfparser.baseimage
         assert base_img.startswith('fedora')
 
+    def test_get_parent_env(self, tmpdir):
+        tmpdir_path = str(tmpdir.realpath())
+        p_env = {"bar": "baz"}
+        df1 = DockerfileParser(tmpdir_path, env_replace=True, parent_env=p_env)
+        df1.lines = [
+            "FROM parent\n",
+            "ENV foo=\"$bar\"\n",
+            "LABEL label=\"$foo $bar\"\n"
+        ]
+
+        assert len(df1.envs) == 2
```

Comment:
Wouldn't the inherited ENVs by definition also need to be available to the child image and therefore would be present? Or am I just completely misunderstanding this?